src/main/java/com/wolox/training/controllers/BookController.java

```diff
@@ -1,18 +1,76 @@
 package com.wolox.training.controllers;
 
-
-import org.springframework.stereotype.Controller;
+import com.wolox.training.exceptions.BookIdMismatchException;
+import com.wolox.training.exceptions.BookNotFoundException;
+import com.wolox.training.models.Book;
+import com.wolox.training.repositories.BookRepository;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.ModelAndView;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    private final BookRepository bookRepository;
+
+    public BookController(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
     @GetMapping("/greeting")
-    public String greeting(@RequestParam(name="name", required=false, defaultValue="Stranger") final String name, Model model) {
+    public final ModelAndView greeting (@RequestParam(name="name", required=false, defaultValue="Stranger") final String name, Model model) {
         model.addAttribute("name", name);
-        return "greeting";
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.setViewName("greeting");
+        return modelAndView;
+    }
+
+    @GetMapping
+    public final Iterable<Book> findAll() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public final Book findOne(@PathVariable Long id) {
```

Comment:
`final` must be the parameters, no the returned type

Reply:
The purpose of adding `final` is to make a variable immutable, so you won't be able to change its value when coding and it will prevent you from making errors.
In this case, you won't change the `id` value in this method, or at least you shouldn't change it, that's why you should use `final`.